lib/auth/permissions.go

```diff
@@ -143,14 +145,16 @@ func (a *authorizer) authorizeRemoteUser(u RemoteUser) (*AuthContext, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// The user is prefixed with "remote-" and suffixed with cluster name with
+	// the hope that it does not match a real local user.
 	user, err := services.NewUser(fmt.Sprintf("remote-%v-%v", u.Username, u.ClusterName))
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	return &AuthContext{
-		// this is done on purpose to make sure user does not match some real local user
-		User:    user,
-		Checker: checker,
+		User:          user,
+		Checker:       checker,
+		IsBuiltinRole: false,
```

Comment:
this variable is redundant, this should be a method on the checker, `checker.IsBuiltin()` otherwise this is a duplicate of the same information already present